xla/stream_executor/stream_executor.h

```diff
@@ -324,6 +324,11 @@ class StreamExecutor {
   // Returns the memory limit in bytes supported by this executor.
   virtual int64_t GetMemoryLimitBytes() const = 0;
 
+  // Returns the total number of compute capable devices.
+  virtual absl::StatusOr<int64_t> GetDeviceCount() {
```

Comment:
There's already an interface defined in xla/stream_executor/platform.h to expose this number.  Platform::VisibleDeviceCount.  There are several places in XLA where this is already used.  It doesn't belong on StreamExecutor or its derived classes.

Reply:
Thanks for the pointer. I updated the pr to use that and removed the new additions to SE.